### DISCUSSION TITLE
shell_execute_cmd() for debugging clarification

### DISCUSSION THREAD

**Discussion by rcarnemollaBE** [zephyrproject-rtos/zephyr#69364] [General]
Why is the shell_execute_cmd() function noted only for use in debugging? I'm hoping to understand why it isn't recommended for other use.
https://docs.zephyrproject.org/apidoc/latest/group__shell__api.html#gabfb6a1f1f53f90365de349348015311e

**Comment by henrikbrixandersen** [comment]
Well, for one the Zephyr shell commands are not governed by the API lifecycle (https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html) process, so their calling signatures can change from release to release without deprecated or other guards in place. This is very difficult to check for in an application.

Second, all functionality exposed by the Zephyr shell modules relies on functionality exposed by proper C APIs (drivers, subsystems, ...). There's no real - or should be no - benefit of calling these through the indirection of the shell.

**Comment by nordicjm** [comment]
Thirdly would probably be security, things like bash have a lot of code to try and enhance security which the zephyr shell does not because the point in it is a debugging feature, not as an actual shell that you'd use in a secure end product
